### ISSUE TITLE
Fix rubocop RSpec/DescribedClassModuleWrapping warnings

### ISSUE THREAD

**Issue by denny** [denny/ShinyCMS-ruby#1111]
I've decided to move this into [the rubocop todo file](https://github.com/denny/ShinyCMS-ruby/blob/main/plugins/ShinyCMS/.rubocop_todo.yml), as something that should be removed from the code at some point.

I think it's a pretty simple fix, but there are about 30 spec files that need changing so it might take a while to work through.

Basic instructions:
1. Remove the exception from the todo file.
2. Run rubocop.
3. For each file this cop warns about:
    a. Remove the module wrapper from the spec file
    b. Fix any class references in that file that need explicit namespacing after 3a

If anybody wants to pick it up, that'd be great (and I'm happy to help, just get in touch if you want a bit more explanation or even to pair on it), otherwise it can sit in the backlog until I want something simple to sort out :)

**Comment by MrBowmanXD** [comment]
hello, i would like to approach this issue but as a beginner rails """developer""" would appreciate some sort of guidance, thanks in advance.

**Comment by denny** [comment]
Hi @MrBowmanXD, thanks for offering to look at this!

It's been ages, but I think the basic idea is that I turned this warning off because there were a lot of places that triggered it and I was too lazy to fix them all 😄 It should mostly be a matter of making the same kind of fix in a bunch of places though, so once you get the idea you should be able to tidy up most of them.

So, step 1 would be to grab the code (probably easiest to create a fork) and get yourself into a new branch on your own machine, and then run `rubocop` to make sure it passes before you change anything - so you know you're starting from a good place. If rubocop doesn't run or something weird happens then drop a comment with the details and we'll try to figure out what's going on.

If your rubocop output looks okay to start with, then open the RuboCop TODO file and remove lines 13 to 16 inclusive:
`https://github.com/denny/ShinyCMS-ruby/blob/main/plugins/ShinyCMS/.rubocop_todo.yml#L13`

Those are the lines that ignore this warning. So, now if you re-run `rubocop`, you should get a load of new warnings for `RSpec/DescribedClassModuleWrapping` - I'm seeing '27 offenses detected' for that on my machine once I take out those lines I mentioned.

Now, you need to change each file that RuboCop complained about in those warnings, to get rid of the complaint. I think this means changing the top two lines that look like this:
```
module ShinyAccess
  RSpec.describe MainSiteHelper, type: :helper do
```
so instead they look like this:
```
RSpec.describe ShinyAccess::MainSiteHelper, type: :helper do
```
...and remove the corresponding closing `end` at the end of the file, and take out one level of indenting throughout.

(This is all because at some point after I originally wrote the tests, rubocop got stricter about how module namespaces should be used in spec files.)

If I remember correctly, there might be a few files that aren't going to be straightforward, so if you can't get one to work don't worry too much - try another file, get the easy ones sorted out first, and then we can take a closer look at anything that's left after that. 🙂 

Drop me a comment if anything's still not clear, or submit a PR if you think you've done it all (or if you've done most of it, and you want to talk about the rest).

**Comment by MrBowmanXD** [comment]
Thank you so much i will start today or tomorrow morning.

**Comment by MrBowmanXD** [comment]
There is two remaining warnings, but wanted to see it with you to not go a bad route with a diferent solution. Thank you for the opportunity by the way.

**Comment by MrBowmanXD** [comment]
Hello Denny, how are you? What about the two remaining warnings?

**Comment by denny** [comment]
Yeah, I'm not sure why the toolchain didn't highlight those for me, something for me to look into. I've added the exception config back in for them while I look at it a bit more. Good progress so far though!

**Comment by MrBowmanXD** [comment]
Cool, happy to help, wanted to checkout others issues, or if you want i can delve deeper in the two remaining warnings (had some trouble solving them).

**Comment by denny** [comment]
Find something else to play with for now, I suspect the solution here might not be straightforward - it seems to have got tangled with an unrelated bug possibly. I'll have a dig around and let you know what I find once I've made sense of it all 🙂

**Comment by MrBowmanXD** [comment]
Cool i will check out other issues

**Comment by denny** [comment]
Ah, no - it was something straightforward after all 😅

Here's the commit where I merged your PR:
https://github.com/denny/ShinyCMS-ruby/commit/f5fc17f4886ca7e08eea04324d0d2626f7c7743c
https://app.circleci.com/pipelines/github/denny/ShinyCMS-ruby/5827

You can see two rubocop failures and one rspec failure there.

First I added the exception config back in for the two trickier bits of rubocop stuff (I haven't looked into fixing those yet, I just set rubocop to ignore them again):
https://github.com/denny/ShinyCMS-ruby/commit/da21bb8f697a53f1a3432630bbe779307f589082
https://app.circleci.com/pipelines/github/denny/ShinyCMS-ruby/5828

You can see that's down to just a single rspec error there now.

And then I spotted and fixed the namespacing error I introduced when I merged your PR:
https://github.com/denny/ShinyCMS-ruby/commit/67b7ee1150068e14a781edc615de69718eb23049
https://app.circleci.com/pipelines/github/denny/ShinyCMS-ruby/5829

Basically because I'd changed how modules were namespaced there, I had to specify the full module name (ShinyForms::Form) on line 69, rather than just the local namespace version (Form). Once I did that everything passed rubocop and rspec again.

Does that all make sense now? Thanks again for the contribution! And I hope the aftermath was more educational than it was confusing 😄

**Comment by MrBowmanXD** [comment]
Took two reads but i think i understood. It makes sense, you explain pretty well. It was pretty interesting and educational tackling this issue. 😇
